summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -196,9 +216,31 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
       StormStatProvider.registerMetrics(jobID, context, countersForSpout)
       SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
     })
-    val stormSpout = tormentaSpout.openHook(registerAllMetrics.get).getSpout
-    val parallelism = getOrElse(stormDag, node, parOpt.getOrElse(DEFAULT_SOURCE_PARALLELISM)).parHint
-    topologyBuilder.setSpout(nodeName, stormSpout, parallelism)
+    tormentaSpout = tormentaSpout.openHook(registerAllMetrics.get)
+
+    val summerOpt: Option[SummerNode[Storm]] = stormDag.dependantsOf(node.asInstanceOf[StormNode]).collect { case s: SummerNode[Storm] => s }.headOption
+    val stormSpout = summerOpt match {
+      case Some(s) => {
+        def GetOrElse[T <: AnyRef: Manifest](default: T, queryNode: StormNode = node) = getOrElse(stormDag, queryNode, default)
+        val summerParalellism = GetOrElse(DEFAULT_SUMMER_PARALLELISM, s)
+        val summerBatchMultiplier = GetOrElse(DEFAULT_SUMMER_BATCH_MULTIPLIER, s)
+        val keyValueShards = executor.KeyValueShards(summerParalellism.parHint * summerBatchMultiplier.get)
+
+        val summerProducer = s.members.collect { case s: Summer[_, _, _] => s }.head.asInstanceOf[Summer[Storm, K, V]]
+        val batcher = summerProducer.store.mergeableBatcher
+        val kvinj = new KeyValueInjection[Int, CMap[(K, BatchID), (Timestamp, V)]]
+        val formattedSummerSpout = tormentaSpout.map {
+          case (time, (k: K, v: V)) =>
+            val newK = keyValueShards.summerIdFor(k)
+            val m = CMap((k, batcher.batchOf(time)) -> (time, v))
+            kvinj((newK, m))
+        }
+        implicit val valueMonoid: Semigroup[V] = summerProducer.semigroup
```

Comment:
can you comment where this is used? I don't see it.

Also, it looks like there is no "map-side" aggregation here, is that correct? Can we add a comment to that effect and explain the design tradeoffs?